packages/client/src/Components/PaperEdits/PaperEdit/ProgramScript/PreviewCanvas/VideoContextProgressBar.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable template-curly-spacing */
+/* eslint-disable react/prop-types */
+import React from 'react';
+import styles from './VideoContextProgressBar.module.css';
+
+class VideoContextProgressBar extends React.PureComponent {
+  constructor(props) {
+    const {
+      width = 640,
+    } = props;
+
+    super(props);
+
+    this.width = width;
+    this.state = { progress: 0 };
+  }
+
+  componentDidUpdate() {
+    this.videoContext = this.props.videoContext;
+    this.setState({ duration: this.videoContext.duration });
```

Comment:
You're at risk of an infinite re-render loop here I think https://reactjs.org/docs/react-component.html#componentdidupdate

May be easier to read the duration directly from props (see those docs ^)

Reply:
also means you won't need to go `this.videoContext && this.videoContext....` as you know `this.props.videoContext` is always there